src/Core/Houston.vala

```diff
@@ -17,85 +17,154 @@
-            try {
-                var res = process_response ((string) mess.response_body.data);
-                if (res.has_member ("data")) {
-                    var data = res.get_array_member ("data");
+            if (root.has_member ("data")) {
+                var data = root.get_array_member ("data");
+                string[] app_ids = {};
+                data.get_elements ().foreach ((id) => {
+                    app_ids += id.get_string ();
+                });
 
-                    foreach (var id in data.get_elements ()) {
-                        app_ids += ((string) id.get_value ());
-                    }
-                }
-            } catch (Error e) {
-                stderr.printf ("Houston: %s\n", e.message);
+                cache_file.set_string_list (category, APPS_KEY, app_ids);
             }
-            Idle.add (get_newest.callback);
-        });
 
-        yield;
-        return app_ids;
+        } catch (Error e) {
+            throw e;
+        }
     }
 
-    public async string[] get_updated () {
-        var uri = HOUSTON_API_URL + "/newest/release";
-        string[] app_ids = {};
+    private async void refresh () {
+        var uri = HOUSTON_API_URL + "/newest/project";
+        debug ("Requesting newest applications from %s", uri);
+        try {
+            yield process_apps (uri, NEWEST_CATEGORY);
+        } catch (Error e) {
+            critical (e.message);
+        }
 
+        uri = HOUSTON_API_URL + "/newest/release";
         debug ("Requesting recently updated applications from %s", uri);
+        try {
+            yield process_apps (uri, UPDATED_CATEGORY);
+        } catch (Error e) {
+            critical (e.message);
+        }
 
-        var message = new Soup.Message ("GET", uri);
-        session.queue_message (message, (sess, mess) => {
+        lock (cache_file) {
             try {
-                var res = process_response ((string) mess.response_body.data);
-                if (res.has_member ("data")) {
-                    var data = res.get_array_member ("data");
-
-                    foreach (var id in data.get_elements ()) {
-                        app_ids += ((string) id.get_value ());
-                    }
-                }
+                cache_file.save_to_file (get_cache_file_path ());
             } catch (Error e) {
-                stderr.printf ("Houston: %s\n", e.message);
+                critical (e.message);
             }
-            Idle.add (get_updated.callback);
+        }
+
+        refreshed ();
+        uint timeout = (uint)(GLib.TimeSpan.DAY/GLib.TimeSpan.MILLISECOND);
+        GLib.Timeout.add (timeout, () => {
+            refresh.begin ();
+            return GLib.Source.REMOVE;
         });
+    }
 
-        yield;
-        return app_ids;
+    private static string get_cache_file_path () {
+        return GLib.Path.build_filename (GLib.Environment.get_user_cache_dir (), "io.elementary.appcenter", "houston.conf");
```

Comment:
About the extension, the file is already in the .cache folder, I'm just using the commonly used extension for keyfiles, I could also remove the extension.

It could go into a static_construct function, but that would also mean that it would take more memory when the cache is only refreshed periodically. So here I've chosen the static function, but we can discuss it further if you wish